### PR TITLE
DM-24920: improve butler CLI mocking

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: tests/test_cliCmdTestIngest.py [hunk 1]
```diff
@@ -55,15 +59,16 @@ def run_test(self, inputs, expectedKwargs):
...
         result = runner.invoke(butler.cli, inputs)
         self.assertEqual(result.exit_code, 0, f"output: {result.output} exception: {result.exception}")
-        verifyFunctionInfo(self, result.output, "ingestRaws", (), expectedKwargs)
+        lsst.obs.base.cli.test.ingestRaws.mock.assert_called_with(**expectedKwargs)
 
     def test_repoAndOutput(self):
         """Test the most basic required arguments, repo and output run"""
         expected = makeExpectedKwargs(repo="here", output_run="out")
-        self.run_test(["ingest-raws", "here", "--output-run", "out"], expected)
+        self.run_test(["ingest-raws", "here",
```
Comment (timj) [inline tests/test_cliCmdTestIngest.py:70]: Is this change needed?

File: tests/test_cliCmdTestIngest.py [hunk 2]
```diff
@@ -43,6 +44,9 @@ def makeExpectedKwargs(**kwargs):
 
 class Suite(unittest.TestCase):
 
+    def setUp(self):
+        sys.modules["lsst.obs.base.script.ingestRaws"] = lsst.obs.base.cli.test.ingestRaws
```
Comment (timj) [inline tests/test_cliCmdTestIngest.py:48]: I'd feel better if there was a teardown that reversed this since `sys.modules` is a global.

File: tests/test_cliCmdTestIngest.py (hunk 1 above)
Comment (n8pease) [reply tests/test_cliCmdTestIngest.py:70]: it's not necessary, but makes this test function look like the rest of the test functions in the suite, making it so that in the tuple of parameters each option+value is on a separate line (the first line is consistently a special case, subcommand+repo), which IMO is easier to read.

File: tests/test_cliCmdTestIngest.py (hunk 2 above)
Comment (n8pease) [reply tests/test_cliCmdTestIngest.py:48]: good point. I've used unittest.patch for this recently, I'll look into using it for this situation.
